modules/custom/az_person/az_person.module

```diff
@@ -146,3 +148,41 @@ function az_person_theme() {
     'horizontal_tabs__node__az_person__group_tabs' => ['base hook' => 'horizontal_tabs'],
   ];
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ *
+ * Prepopulate the alphabetical index letter field in az_person where available.
+ */
+function az_person_node_presave(EntityInterface $entity) {
+  if ($entity->bundle() === 'az_person'
+    && $entity instanceof NodeInterface
+    && $entity->hasField('field_az_alphabetical_index')
+    && $entity->hasField('field_az_lname'))
+  {
```

Comment:
Seeing a PHPCS annotation now.  Apparently my suggestion about readability for multiline conditions wasn't a good idea.  Sorry about that.
```suggestion
    && $entity->hasField('field_az_lname')) {
```